src/main/java/io/seqera/tower/agent/Agent.java

```diff
@@ -98,19 +101,30 @@ public static void main(String[] args) throws Exception {
     public void run() {
         try {
             validateParameters();
-            checkTower();
-            connectTower();
             sendPeriodicHeartbeat();
-        } catch (Exception e) {
+            infiniteLoop();
+        } catch (UnrecoverableException e) {
             logger.error(e.getMessage());
```

Comment:
What about keeping only the `catch (Throwable e)` block?

Reply:
Here is where the `UnrecoverableException` are logged. These exceptions have a meaningful message, I don't want to show the stack trace on these cases. It will be less intuitive for the user.

Reply:
The only difference I see with the Throwable one is that this uses `logger.error(e.getMessage());` instead of `logger.error(e.getMessage(), e);`. But again why omitting `, e)`. There useful to track the stack trace

Reply:
Because I'm using `UnrecoverableException` to report known problems. I already known what was wrong and the message is clear. See for example [this one](http://github.com/seqeralabs/tower-agent/blob/c51dcc4023234095a1c2d9060c4b362e69a4ae4f/src/main/java/io/seqera/tower/agent/Agent.java#L238-L238).

I don't want to show a useless stack trace on these errors.

Reply:
I see the point. Nextflow renders the stack trace only in the log file not the stdout. Would not make sense doing the same?

Reply:
Currently there is no log file at Agent. And I recall giving it a try and rolling it back due to some GraalVM native image problem. We have an issue to track this https://github.com/seqeralabs/tower-agent/issues/23

Reply:
Ok, let's consider this for later